app/UserService.ts

```diff
@@ -405,39 +426,59 @@ export class UserService extends ChordNode {
     }
   }
 
-  migrateUsersToPredecessorOrSuccessor(migrateToPredecessor: boolean) {
-    if (this.objectIsEmpty(this.userMap)) {
-      return null;
-    }
+  migrateUsersToPredecessor() {
+    if (this.userMapIsEmpty()) return null;
+
+    const client = connect(this.predecessor);
 
-    let usersToMigrate;
-    if (migrateToPredecessor) {
-      usersToMigrate = Object.keys(this.userMap).filter(userId =>
+    Object.keys(this.userMap)
+      .filter(hashedKey =>
         isInModuloRange(
-          parseInt(userId, 10),
+          parseInt(hashedKey, 10),
           this.id,
           false,
           this.predecessor.id,
           true
         )
-      );
-    } else {
-      usersToMigrate = Object.keys(this.userMap);
-    }
+      )
+      .forEach(hashedKey => {
+        try {
+          console.log(this.userMap[hashedKey]);
+          client.insertUserRemoteHelper({
+            user: this.userMap[hashedKey],
+            edit: false
+          });
+          this.removeUser(hashedKey);
+        } catch (error) {
+          handleGRPCErrors(
+            "migrateUsersToPredecessor",
+            "insertUserRemoteHelper",
+            this.predecessor.host,
+            this.predecessor.port,
+            error
+          );
+        }
+      });
+    return null;
+  }
+
+  migrateUsersToSuccessor() {
+    if (this.userMapIsEmpty()) return null;
 
-    const client = migrateToPredecessor
-      ? connect(this.predecessor)
-      : connect(this.fingerTable[0].successor);
-    usersToMigrate.forEach(userId => {
+    const client = connect(this.fingerTable[0].successor);
+
+    // Why do we migrate all keys to the successor?
```

Comment:
That's the algorithm.  The successor always includes all keys with values (predecessor, successor]; i.e., from the "predecessor + 1" - included - to the successor itself - included.  So, let's say that you have an ID of "3", which is the maximum value allowed to be stored in a node with ID of "3", and you crush that node "3".  By definition, whatever the next node is (i.e., {3}.successor) would be algorithmically the right place to store from ID "3" on back - until pred+1.